scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -977,17 +978,37 @@ void pingTaskNode(RunningTaskData taskData) {
         } catch (Throwable t) {
             RunningTaskData runningTask = jobs.getRunningTask(task.getId());
             if (runningTask != null) {
-                int attempts = runningTask.increaseAndGetPingAttempts();
-                if (attempts > PASchedulerProperties.SCHEDULER_NODE_PING_ATTEMPTS.getValueAsInt()) {
-                    tlogger.info(task.getId(), "node failed", t);
-                    restartTaskOnNodeFailure(task);
+                // We try to ping the node where the task is running.
+                // We don't consider here other nodes reserved for the task, 
+                // as it is the responsibility of the task itself to manage extra nodes lifecycle 
+                // in case of complex multinodes task deployment.
+                if (!pingNode(runningTask.getNodes().get(0))) {
```

Comment:
Increasing timeouts does not solve the problem of false positives when the TaskLauncher kills itself at the end of the task. As in this case, the exception raised will be that the TaskLauncher Active Object does not exist (it's not a timeout issue).

Basically here, there are two responsibilities. The first responsibility of this method is to get the progress of the task and store it. If there is an error while getting this progress (which can happen for several reasons, such as node failure or task ending), then we switch to the second responsibility which is node failure detection. I will create more explicit submethods to make things clearer.